backend/ownerSizeTotalWorker.test.js

```diff
@@ -0,0 +1,288 @@
+      await appendToIndexFactory(`_private_indirectResources_${indirectOwnerID}`)(resourceID)
+    }
+  }
+
+  static registerBillableEntity = appendToIndexFactory('_private_billable_entities')
+
+  static async saveOwner (ownerID: string, resourceID: string) {
+    // Store the owner for the current resource
+    // Use a queue to check that the owner exists, preventing the creation of
+    // orphaned resources (e.g., because the owner was just deleted)
+    await sbp('chelonia.db/set', `_private_owner_${resourceID}`, ownerID)
+    const resourcesKey = `_private_resources_${ownerID}`
+    // Store the resource in the resource index key
+    // This is done in a queue to handle several simultaneous requests
+    // reading and writing to the same key
+    await appendToIndexFactory(resourcesKey)(resourceID)
+    setTimeout(() => this.saveIndirectResourcesIndex(resourceID), (0, Math.random)() * 500 | 0)
+  }
+
+  async init () {
+    await updateSize(this.id, `_private_size_${this.id}`, this.ownSize)
+    if (this.owner) {
+      await this.constructor.saveOwner(this.owner.id, this.id)
+    } else {
+      await this.constructor.registerBillableEntity(this.id)
+    }
+  }
+
+  async newMessage () {
+    const size = (0, Math.random)() * 9999 | 0
+    this.ownSize += size
+    await updateSize(this.id, `_private_size_${this.id}`, size)
+  }
+
+  async newKvValue () {
+    const val = (0, Math.random)() * 9999 | 0
+    const sign = (0, Math.random)()
+    const size = (sign < 0.5) ? Math.max(-this.kvSize, -val) : val
+    this.kvSize += size
+    await updateSize(this.id, `_private_size_${this.id}`, size)
+  }
+}
+
+async function * randomOp (iterations) {
+  const mainContracts: Contract[] = []
+  const allContracts: Contract[] = []
+
+  {
+    const contract = new Contract()
+    console.log('Created initial contract', contract.id)
+    await contract.init()
+    mainContracts.push(contract)
+    allContracts.push(contract)
+    yield
+  }
+
+  while (iterations-- > 0) {
+    const op = (0, Math.random)() * 9 | 0
+    switch (op) {
+      case 0: {
+        const contract = new Contract()
+        console.log('Created new main contract', contract.id)
+        await contract.init()
+        mainContracts.push(contract)
+        allContracts.push(contract)
+        break
+      }
+      case 1:
+      case 2:
+      case 3:
+      case 4:
+      case 5: {
+        if (!allContracts.length) break
+        const idx = (0, Math.random)() * allContracts.length | 0
+        const contract = allContracts[idx]
+        const subop = (0, Math.random)() * 4 | 0
+        switch (subop) {
+          case 0: {
+            console.log('Simulated new event on', contract.id)
+            await contract.newMessage()
+            break
+          }
+          case 1: {
+            console.log('Simulated KV update on', contract.id)
+            await contract.newKvValue()
+            break
+          }
+          case 3: {
```

Comment:
Did you mean to skip `case 2`? If so could you add a comment as to why?

Reply:
Yes, I believe that was to do nothing. I agree it looks strange.